Java/cp/NQueens.java

```diff
@@ -0,0 +1,61 @@
+/**
+N-Queens problem is a famous problem
+The paradigm used to solve the problem is backtracking
+**/
+
+import java.io.*;
+import java.util.*;
+
+public class NQueens {
+    
+  public static void solve(boolean[][] board, boolean[] cols, 
+    boolean[] ndiag, boolean[] rdiag, int row, String asf) {
+        
+        if(row == board.length) {
+            System.out.println(asf+'.');
+            return;
+        }
+        
+        for(int col=0; col<board.length; col++) {
+            if(cols[col] == false && ndiag[row+col] == false && 
+                rdiag[row-col+board.length-1] == false) {
+                    cols[col] = true;
+                    ndiag[row+col] = true;
+                    rdiag[row-col+board.length-1] = true;
+                    board[row][col] = true;
+                    solve(board, cols, ndiag, rdiag, row+1, asf+row+'-'+col+", ");
+                    cols[col] = false;
+                    ndiag[row+col] = false;
+                    rdiag[row-col+board.length-1] = false;
+                    board[row][col] = false;
+                }
+        }
+    }
+
+  public static void main(String[] args) throws Exception {
+    Scanner scn = new Scanner(System.in);
+    int n = scn.nextInt();
+    boolean[][] board = new boolean[n][n];
+    //write your code here
```

Comment:
This Comment is redundant. If you are solving a CP Problem from an Online Portal, prefer to add the link as well.